src/pages/spawn/spawnVolume/spawnVolumeTable/spawnVolumeTableActions/MigrateBtn.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+import Button, { Size } from "@leafygreen-ui/button";
+import { SpawnHostModal } from "pages/spawn/spawnHost/SpawnHostModal";
+import { MyVolume } from "types/spawn";
+
+interface Props {
+  volume: MyVolume;
+}
+
+export const MigrateBtn: React.VFC<Props> = ({ volume }) => {
+  const [openModal, setOpenModal] = useState(false);
+
+  return (
+    <>
+      <Button
+        size={Size.XSmall}
+        data-cy={`migrate-btn-${volume.displayName || volume.id}`}
+        onClick={(e) => {
+          e.stopPropagation();
+          setOpenModal(true);
+        }}
+      >
+        Migrate
+      </Button>
+      <SpawnHostModal
```

Comment:
We should probably extract the shared logic into a separate component and have a spawn host component and migrate component that abstracts the shared logic.

Reply:
+1, would allow for more flexibility in the future and also facilitate converting the `EditSpawnHostModal` component to use a SpruceForm. There are a few other changes to make for the migration modal:

- Only allow selection of virtual workstation distros as per our Slack discussion
- Change the modal heading to match designs
- Add some modal help text: "Migrate this home volume to a new host. Upon successful migration, the unused host will be scheduled to expire in 24 hours."